### PR TITLE
Broken title search ut

### PR DESCRIPTION
**Pull request by nadolskit**
**Summary of Problem**
The expected vs asserted citation count was not matching, causing this UT to specifically fail. 

I noticed if I switched the order of sources from `semantic_scholar, crossref` to `crossref, semantic_scholar` the citation count would be even more dramatically different (similar to the results James was noticing earlier)

More importantly, I found the papers on [semantic scholar](https://www.semanticscholar.org/paper/PaperQA%3A-Retrieval-Augmented-Generative-Agent-for-L'ala-O'Donoghue/7e55d8701785818776323b4147cb13354c820469) (25); which doesn't match our [expected value](https://github.com/Future-House/paper-qa/blob/main/tests/test_clients.py#L69) (23); so that failure makes sense.
The following assertion is failing for the same reason: [196](https://www.semanticscholar.org/paper/Augmenting-large-language-models-with-chemistry-Bran-Cox/354dcdebf3f8b5feeed5c62090e0bc1f0c28db06) assertion vs [191](https://github.com/Future-House/paper-qa/blob/main/tests/test_clients.py#L93) expected.

Vibe checked the results of this change via a CI run: https://github.com/Future-House/paper-qa/actions/runs/10859898582/job/30139826477

This does not fix the issue that James found earlier this evening in test_agents.py.

### CODE REVIEW

File: tests/utils/paper_helpers.py
Comment (jamesbraza) [inline tests/utils/paper_helpers.py:7]: We use Google Style docstrings, you'll perhaps want to update your IDE

File: tests/utils/paper_helpers.py
Comment (jamesbraza) [inline tests/utils/paper_helpers.py:16]: I think you got the capturing groups wrong, you want `r"This article has (\d+) citations?"`

We don't care to extract the text, we want to extract the number

Then you can just directly compare the `int`, no need for the weird `r"\1n\2"`

File: tests/utils/paper_helpers.py
Comment (jamesbraza) [inline tests/utils/paper_helpers.py:12]: This regex is closely coupled to `DocDetails`. Can you move this regex to a `ClassVar[str]` on `DocDetails`?

File: tests/utils/paper_helpers.py
Comment (nadolskit) [reply tests/utils/paper_helpers.py:16]: That's fair. 
This original approach using `\1n\2` was because it's focusing on comparing the structure of the citation text while ignoring the citation count itself; just ignoring whatever number is there.

But you make a valid point. I'll get this updated!

Comment (jamesbraza) [reply tests/utils/paper_helpers.py:16]: Fwiw (I am not sure your experience levels with regex) going from `r"(This article has )\d+( citations?)"` to `r"This article has (\d+) citations?"` will still match the surrounding text too. A capturing group `()` is here to make it easy to capture and extract a value

And plz point out if we're on the same page already haha. Hope you have a good night

File: tests/utils/paper_helpers.py
Comment (nadolskit) [reply tests/utils/paper_helpers.py]: done!

File: tests/utils/paper_helpers.py
Comment (nadolskit) [reply tests/utils/paper_helpers.py]: I believe I've addressed this accordingly. Thx James!

File: tests/utils/paper_helpers.py
Comment (nadolskit) [reply tests/utils/paper_helpers.py]: No longer required

**Comment by mskarlin** [comment]
The minor variations, i.e. 196 vs 191 is expected, because the papers will get more citations over time. The bigger question is why the cassettes aren’t being used because this should come back with a recorded request each time, I’m in favor of root causing that before we merge this. Maybe we regenerated these cassettes but didn’t update the test?

File: tests/test_clients.py
Comment (jamesbraza) [inline tests/test_clients.py:133]:
```suggestion
        assert (
            expected_citation_format
        ), "Expected string should match the citation pattern"
        assert (
            actual_citation_format
        ), "Actual string should match the citation pattern"
```

This may resolve the `union-attr` ignore a few lines below

File: tests/test_clients.py
Comment (jamesbraza) [inline tests/test_clients.py:148]: So we are doing a regex search to find a region, then removing that region, and lastly doing an equality check on the front/back end.

I think it would be simpler to do one regex removal of the region, and directly equality compare the remanants.

I also think this is what you originally had 😅 😓  hahaha sorry! I actually realized I misunderstood your original logic last night, my bad here.

What do you think?

File: tests/test_clients.py
Comment (jamesbraza) [inline tests/test_clients.py:154]:
```suggestion
            # Equality check all attributes but the ones in the below set
            if key not in {"is_oa", "source", "formatted_citation"}:
```

Just making this logic easier to understand

File: tests/test_clients.py
Comment (nadolskit) [reply tests/test_clients.py:148]: definitely! no worries, communicating through PRs can be hard sometimes. Thank you!! adjusting now.

before I merge this I'm also going to look into why the cassette file aren't (hasn't) been updated

File: paperqa/types.py
Comment (jamesbraza) [inline paperqa/types.py:655]: Can you just add a comment explaining the race condition-y thing you encountered that this fixes, either on this line or just above this `if` clause

File: paperqa/types.py
Comment (jamesbraza) [inline paperqa/types.py:667]:
```suggestion
                    merged_data[field] = (
                        self_value
                        if self_value is not None  # Dance around 0
                        else other_value
                    )
```

**Review by jamesbraza — APPROVED** [review]
Not sure why some requests were missing from fixtures, but otherwise LGTM and thanks for doing this

**Pull request merged**
